src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,343 @@
+    private String vidMapJSONFileName = DEFAULT_VIDMAP_FILE_NAME;
+
+    @Argument(fullName = OVERWRITE_WORKSPACE_NAME,
+              shortName = OVERWRITE_WORKSPACE_NAME,
+              doc = "Will overwrite given workspace if it exists. " +
+                    "Otherwise a new workspace is created. " +
+                    "Defaults to false",
+              optional = true)
+    private Boolean overwriteExistingWorkspace = false;
+
+    @Override
+    public boolean requiresIntervals() { return true; }
+
+    @Override
+    public boolean requiresReads() { return false; }
+
+    @Override
+    public boolean requiresReference() {
+      return false;
+    }
+
+    // The GenomicsDB importer object
+    private GenomicsDBImporter importer = null;
+
+    // Intervals from command line (singleton for now)
+    private List<ChromosomeInterval> intervals;
+
+    private Map<String, FeatureReader<VariantContext>> sampleToVCFMap = new HashMap<>();
+
+    // Needed as smartMergeHeaders() returns a set of VCF header lines
+    private Set<VCFHeaderLine> mergedHeader = null;
+
+    // VCFHeader object created from mergedHeader
+    private VCFHeader mergedVCFHeader = null;
+
+    /**
+     * Before traversal, fix configuration parameters and initialize
+     * GenomicsDB. Hard-coded to handle only VCF files and headers
+     */
+    @Override
+    public void onTraversalStart() {
+
+        File workspaceDir = checkOrCreateWorkspace();
+
+        if (vcfBufferSizePerSample < 1024L) {
+            logger.warn("Size per column partition per sample is tool small." +
+              " Consider using larger size than 10KBytes");
+        }
+
+        File vidMapJSONFile = (vidMapJSONFileName.equals(DEFAULT_VIDMAP_FILE_NAME)) ?
+            new File(workspaceDir + "/" + vidMapJSONFileName) :
+            new File(vidMapJSONFileName);
+
+        File callsetMapJSONFile = (callsetMapJSONFileName.equals(DEFAULT_CALLSETMAP_FILE_NAME)) ?
+            new File (workspaceDir + "/" + callsetMapJSONFileName) :
+            new File(callsetMapJSONFileName);
+
+        logger.info("Vid Map JSON file will be written to " + vidMapJSONFile);
+        logger.info("Callset Map JSON file will be written to " + callsetMapJSONFile);
+        
+        long variantContextBufferSize = vcfBufferSizePerSample * sampleToVCFMap.size();
+
+        logger.info("Writing data to array - " + workspace + "/" + arrayName);
+
+        try {
+            logger.info("GenomicsDB import batch started");
+            importer = new GenomicsDBImporter(sampleToVCFMap, mergedHeader, intervals.get(0), workspace,
+                                              arrayName, variantContextBufferSize, segmentSize,
+                                              vidMapJSONFile.getAbsolutePath(),
+                                              callsetMapJSONFile.getAbsolutePath());
+        } catch (IOException e) {
+            throw new UserException("GenomicsDB constructor has errors: " + e.getMessage());
```

Comment:
Error message is not user-friendly -- change to "Error initializing GenomicsDBImporter"